### PR TITLE
Return the key in s3_iter_bucket

### PR DESCRIPTION
**Pull request by salilb**
Returns the entire key instead of only the key name. Breaks compatibility with older code!!!

### CODE REVIEW

**Comment by piskvorky** [comment]
Travis build fails -- will require a change to unit tests as well, I suppose.

**Comment by salilb** [comment]
Do you want to make changes to the unit tests? Else, I'll try to fix it later.

Could returning the whole key have an impact on the performance?

**Comment by piskvorky** [comment]
Is the key object pickleable (can be `pickle.dump()`ed)?

**Comment by salilb** [comment]
I fixed the tests. Not the most elegant solution but it passes the tests.

File: smart_open/tests/test_smart_open.py [hunk 1]
```diff
@@ -462,11 +462,17 @@ def test_s3_iter_bucket_moto(self):
                 expected[key_name] = content
 
         # read all keys + their content back, in parallel, using s3_iter_bucket
-        result = dict(smart_open.s3_iter_bucket(mybucket))
+        result = {}
+        for k, c in smart_open.s3_iter_bucket(mybucket):
+            result[k.name] = c
+        #result = dict(smart_open.s3_iter_bucket(mybucket))
```
Comment (piskvorky) [inline smart_open/tests/test_smart_open.py]: We don't want to keep dead code around, even as comments. The line is preserved in git history, if we ever need it again.

File: smart_open/tests/test_smart_open.py [hunk 2]
```diff
@@ -462,11 +462,17 @@ def test_s3_iter_bucket_moto(self):
...
         self.assertEqual(expected, result)
 
         # read some of the keys back, in parallel, using s3_iter_bucket
-        result = dict(smart_open.s3_iter_bucket(mybucket, accept_key=lambda fname: fname.endswith('4')))
+        result = {}
+        for k, c in smart_open.s3_iter_bucket(mybucket, accept_key=lambda fname: fname.endswith('4')):
+            result[k.name] = c
+        #result = dict(smart_open.s3_iter_bucket(mybucket, accept_key=lambda fname: fname.endswith('4')))
```
Comment (piskvorky) [inline smart_open/tests/test_smart_open.py]: ditto

**Comment by piskvorky** [comment]
Looks elegant enough to me :) Please fix the dead code and we can merge :+1:

**Comment by salilb** [comment]
done! :)

**Comment by piskvorky** [comment]
Merged, thanks @salilb !

**Pull request merged**